Bottom: c83a07466a1c73a53d23ecb1cd50fffda2f752b8
Top:    fd2be63602554b0fb762bd95a5a95fc3f594be74
Author: Ian Kent <ikent@redhat.com>
Date:   2013-09-23 16:51:10 +0800

Refresh of debug.patch

---

diff --git a/lib/rpc_subs.c b/lib/rpc_subs.c
index c7493cf..fb79eb0 100644
--- a/lib/rpc_subs.c
+++ b/lib/rpc_subs.c
@@ -44,21 +44,9 @@
 #endif
 
 #ifdef WITH_LIBTIRPC
-static const char *rpcb_pgmtbl[] = {
-	"rpcbind", "portmap", "portmapper", "sunrpc", NULL,
-};
-static const char *rpcb_netnametbl[] = {
-	"rpcbind", "portmapper", "sunrpc", NULL,
-};
 const rpcprog_t rpcb_prog = RPCBPROG;
 const rpcvers_t rpcb_version = RPCBVERS;
 #else
-static const char *rpcb_pgmtbl[] = {
-	NULL,
-};
-static const char *rpcb_netnametbl[] = {
-	NULL,
-};
 const rpcprog_t rpcb_prog = PMAPPROG;
 const rpcvers_t rpcb_version = PMAPVERS;
 #endif
@@ -345,6 +333,9 @@ static pthread_mutex_t rpcb_mutex = PTHREAD_MUTEX_INITIALIZER;
 static rpcprog_t rpc_getrpcbyname(const rpcprog_t program)
 {
 #ifdef HAVE_GETRPCBYNAME
+	static const char *rpcb_pgmtbl[] = {
+		"rpcbind", "portmap", "portmapper", "sunrpc", NULL,
+	};
 	struct rpcent *entry;
 	unsigned int i;
 
@@ -364,6 +355,9 @@ static rpcprog_t rpc_getrpcbyname(const rpcprog_t program)
 static unsigned short rpc_getrpcbport(const int proto)
 {
 #ifdef HAVE_GETSERVBYNAME
+	static const char *rpcb_netnametbl[] = {
+		"rpcbind", "portmapper", "sunrpc", NULL,
+	};
 	struct servent *entry;
 	unsigned int i;
